content/Methods/Methods.q.md

```diff
@@ -0,0 +1,1329 @@
+        }
+    }
+    $pObject = new Person();
+    $pObject->changeName("Charles");
+    $pObject->display;
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+
+        public function changeName($newName)
+        {
+            $this->name = $newName;
+        }
+
+        public function display()
+        {
+            echo "My name is " . $this name . ".";
+        }
+    }
+    $pObject = new Person();
+    $pObject->changeName("Charles");
+    $pObject->display();
+?>
+```
+/// type=SS, answer=[3]
+
+Execute the program. What is the error message?
+
+ - Use of undefined constant `name` - assumed `'name'` on line number 13
+
+ - Object of class `Person` could not be converted to string on line number 13
+
+ - syntax error, unexpected `'name'` (T_STRING), expecting `','` or `';'` on line number 13
+
+ - syntax error, unexpected `'$this'` (T_VARIABLE), expecting `','` or `';'` on line number 13
+
+ - syntax error, unexpected `'"."'` (T_CONSTANT_ENCAPSED_STRING), expecting `','` or `';'` on line number 13
+
+
+/// type=MS, answer=[1,3]
+
+Which statements correctly describe the error?
+
+ - On line 13, `$this name` in the `echo` statement is invalid.
+
+ - On line 13, `$this name` is not enclosed in double quotes `""`.
+
+ - There is no object operator `->` between `$this` and `name` on line 13.
+
+ - On line 13, the `this` pseudo-variable does not start with a dollar sign `$`.
+
+ - There is no concatenation operator `.` between `$this` and `name` on line 13.
+
+:::
+
+
+/// type=CR, answer=[tests/Methods/MissingObjectOperatorTest.php]
+
+Correct the code so that it outputs the string `My name is Charles.`.
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+
+        public function changeName($newName)
+        {
+            $this->name = $newName;
+        }
+
+        public function display()
+        {
+            echo "My name is " . $this name . ".";
+        }
+    }
+    $pObject = new Person();
+    $pObject->changeName("Charles");
+    $pObject->display();
+?>
+```
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+
+        public function changeName($newName)
+        {
+            $this->name = $newName;
+        }
+
+        public function display()
+        {
+            echo "My name is " . $this->name . ".";
+        }
+    }
+    $pObject = new Person();
+    $pObject->changeName("Charles");
+    $pObject->display();
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+ - It prints `Charles`.
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `My name is Diana.`.
+
+ - It prints `My name is Charles.`.
+
+
+/// type=MS, answer=[2,3]
+
+Remove the object operator `->` from the statement `$pObjectdisplay();` on line 18. Execute the program. What are the error messages?
```

Comment:
there's no `$pObjectdisplay();` in line 18, only `$pObject->display();`

Reply:
My mistakes, sorry. This is supposedly `$pObject->display();`. Thank you for pointing this out.